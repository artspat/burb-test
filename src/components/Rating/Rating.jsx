import React from 'react';
import cx from 'classnames';
import StarRatings from 'react-star-ratings';
import './Rating.scss';

export const Rating = ({ className = '', value = 0 }) => {
    let clarifiedValue = value;
    if (value > 10) {
        clarifiedValue = 10;
    }
    if (value < 0) {
        clarifiedValue = 0;
    }
    
    return (
        <div className={cx('b-rating', className)}>
            <StarRatings
                name='rating'
                rating={clarifiedValue}
                starSpacing="1px"
                numberOfStars={10}
                starDimension="16px"
                starRatedColor="red"
            />
            <span className="b-rating__value">{clarifiedValue}</span>
        </div>
    );
};
