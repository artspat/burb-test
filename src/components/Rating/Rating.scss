.b-rating {
    &__value {
        margin-left: 7px;
        font-weight: bold;
    }
}
