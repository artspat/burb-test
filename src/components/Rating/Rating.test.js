import React from 'react';
import { shallow } from 'enzyme';
import { Rating } from './Rating';

describe('<Rating />', () => {
    describe('when props are passed', () => {
        it('should render component', () => {
            const wrapper = shallow(<Rating className="css-class" value={3} />);
            expect(wrapper).toMatchSnapshot();
        });
    });

    describe('when props are not passed', () => {
        it('should use default values', () => {
            const wrapper = shallow(<Rating />);
            expect(wrapper).toMatchSnapshot();
        });
    });

    describe('when value is less than 0', () => {
        it('should use 0 as value', () => {
            const wrapper = shallow(<Rating value={-5} />);
            expect(wrapper).toMatchSnapshot();
        });
    });

    describe('when value is higher than 10', () => {
        it('should use 10 as value', () => {
            const wrapper = shallow(<Rating value={15} />);
            expect(wrapper).toMatchSnapshot();
        });
    });
});
