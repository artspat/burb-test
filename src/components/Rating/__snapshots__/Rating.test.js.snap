// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<Rating /> when props are not passed should use default values 1`] = `
<div
  className="b-rating"
>
  <StarRatings
    changeRating={null}
    gradientPathName=""
    ignoreInlineStyles={false}
    name="rating"
    numberOfStars={10}
    rating={0}
    starDimension="16px"
    starEmptyColor="rgb(203, 211, 227)"
    starHoverColor="rgb(230, 67, 47)"
    starRatedColor="red"
    starSpacing="1px"
    svgIconPath="m25,1 6,17h18l-14,11 5,17-15-10-15,10 5-17-14-11h18z"
    svgIconViewBox="0 0 51 48"
    typeOfWidget="Star"
  />
  <span
    className="b-rating__value"
  >
    0
  </span>
</div>
`;

exports[`<Rating /> when props are passed should render component 1`] = `
<div
  className="b-rating css-class"
>
  <StarRatings
    changeRating={null}
    gradientPathName=""
    ignoreInlineStyles={false}
    name="rating"
    numberOfStars={10}
    rating={3}
    starDimension="16px"
    starEmptyColor="rgb(203, 211, 227)"
    starHoverColor="rgb(230, 67, 47)"
    starRatedColor="red"
    starSpacing="1px"
    svgIconPath="m25,1 6,17h18l-14,11 5,17-15-10-15,10 5-17-14-11h18z"
    svgIconViewBox="0 0 51 48"
    typeOfWidget="Star"
  />
  <span
    className="b-rating__value"
  >
    3
  </span>
</div>
`;

exports[`<Rating /> when value is higher than 10 should use 10 as value 1`] = `
<div
  className="b-rating"
>
  <StarRatings
    changeRating={null}
    gradientPathName=""
    ignoreInlineStyles={false}
    name="rating"
    numberOfStars={10}
    rating={10}
    starDimension="16px"
    starEmptyColor="rgb(203, 211, 227)"
    starHoverColor="rgb(230, 67, 47)"
    starRatedColor="red"
    starSpacing="1px"
    svgIconPath="m25,1 6,17h18l-14,11 5,17-15-10-15,10 5-17-14-11h18z"
    svgIconViewBox="0 0 51 48"
    typeOfWidget="Star"
  />
  <span
    className="b-rating__value"
  >
    10
  </span>
</div>
`;

exports[`<Rating /> when value is less than 0 should use 0 as value 1`] = `
<div
  className="b-rating"
>
  <StarRatings
    changeRating={null}
    gradientPathName=""
    ignoreInlineStyles={false}
    name="rating"
    numberOfStars={10}
    rating={0}
    starDimension="16px"
    starEmptyColor="rgb(203, 211, 227)"
    starHoverColor="rgb(230, 67, 47)"
    starRatedColor="red"
    starSpacing="1px"
    svgIconPath="m25,1 6,17h18l-14,11 5,17-15-10-15,10 5-17-14-11h18z"
    svgIconViewBox="0 0 51 48"
    typeOfWidget="Star"
  />
  <span
    className="b-rating__value"
  >
    0
  </span>
</div>
`;
